code/datums/controllers/action_controls.dm

```diff
@@ -915,6 +915,12 @@ var/datum/action_controller/actions
 						for (var/mob/O in AIviewers(owner))
 							O.show_message("<span class='alert'><B>...and triggers it accidentally!</B></span>", 1)
 						MT.triggered(source, source.hand ? "l_hand" : "r_hand")
+				else if (istype(I, /obj/item/device/mousepunch))
+					var/obj/item/device/mousepunch/MT = I
+					if (MT.armed)
+						for (var/mob/O in AIviewers(owner))
```

Comment:
The visible_message should only be called once on the source of the message, it does this loop inside of itself.

Reply:
Setting it to src.show_message and src.owner.show_message appears to not cause the message to pop up.

Reply:
That's why I am telling you to use visible_message and not show_message here.